indexing-service/src/main/java/org/apache/druid/indexing/common/task/HadoopIndexTask.java

```diff
@@ -529,7 +529,10 @@ private TaskStatus runInternal(TaskToolbox toolbox) throws Exception
       indexerGeneratorCleanupJob(
           indexGeneratorJobAttempted,
           indexGeneratorJobSuccess,
-          indexerSchema == null ? null : toolbox.getJsonMapper().writeValueAsString(indexerSchema)
+          indexerSchema == null ? null : toolbox.getJsonMapper().writeValueAsString(
+                  indexerSchema.withTuningConfig(
+                          indexerSchema.getTuningConfig().withVersion(
+                                  version == null ? indexerSchema.getTuningConfig().getVersion() : version)))
```

Comment:
Might be nicer to move all of this logic inside the `indexerGeneratorCleanupJob` and just pass the `version` and the `indexerSchema` itself to that method rather than a String.

Reply:
I cleaned it up a bit by handling the schema update with a new version above this in the code. I think it makes things more readable and avoids refactoring the indexerGeneratorCleanupJob. Let me know what you think. I will have to address the coverage in an additional commit once I have a few minutes to look at the tests that exist already